pkg/vendir/directory/named_ref_fetcher.go

```diff
@@ -25,6 +26,9 @@ func (f NamedRefFetcher) GetSecret(name string) (ctlconf.Secret, error) {
 	var found []ctlconf.Secret
 	for _, secret := range f.secrets {
 		if secret.Metadata.Name == name {
+			if len(found) == 1 && reflect.DeepEqual(found[0].Data, secret.Data) {
```

Comment:
code wouldnt work for 3 secrets that are the same.

Reply:
The test was done with 3 secrets, if the second secret has the same `data` as the first one, it won't be added, if the third one has the same `data`, won't be added as well, of one of the secrets with same name has different `data` it will fail with multiple secrets error due to we would not assume which one is good to go

Reply:
you are right. [secret A, secret B, secret B] wont work though, since you are hard coding index lookups.

Reply:
Yes, that is the intent, if there are more than one secret with same name but different `data`, we should fail as we do not know which secret.data to use

Reply:
That seems reasonable to me, I think in cases where there is obscurity - we should defer to the user.

`[secret A, secret B, secret B]` - When you mentioned this @cppforlife , did you have an outcome you were expecting instead of an explicit error from vendir?

Reply:
> [secret A, secret B, secret B] - When you mentioned this @cppforlife , did you have an outcome you were expecting instead of an explicit error from vendir?

im saying the way it's currently implemented, it only deals with a single secret being duplicated, and it has to come in a very specific order (dup-ed secrets must come first). neither of these assumptions should be there. we should be able to support N numbers of dup-ed secrets in any order. to support that, just create a separate map keyed by secret name and do the lookup (dont use arrays for keyed lookups)

Reply:
@cppforlife i am not sure if i am understanding right your concern,
the function only cares of secret by the name, that is why i am comparing only that, and if the `len(found)` is `0` i add the secret, if it is already a secret in `found` with same `data`, we do not need to add it but if there is one secret in `found` (meaning with the name that we care about) but data is different then we add it and later one the `len` validation will throw the error, if there is more than one secrets in `found` then we do no care on validating `data` it will fails due to more than one secret found

Reply:
Perhaps having a test case that fails would be useful here to clear things up? I think in this case being really concrete with a failure will help guide the implementation.

Reply:
i just added the test file with two cases failing (`wantErr: true`)

Reply:
@cppforlife ^ tests were added. If they don't cover what you were expecting, can you add a suggestion for a test that does fail?

Reply:
sorry folks, i screwed up -- i misread and assumed this was a change in a different piece of code [0] that was checking _all_ secrets at the load time, vs doing it on the fly when secret is requested and filtered by name (so that's why `found[0]` is ok). so with current impl "invalid" secret configs wouldn't be errored on until someone uses it. ideally we would check for this condition in [0], but i know yall already put some effort in this area.

[0] https://github.com/carvel-dev/vendir/blob/develop/pkg/vendir/config/config.go#L30

Reply:
It seems to me that is a different issue, correct me if I am wrong, one if allowing the same secret (equal `name` and `data`) multiple times and the other one is checking if secrets are yaml valid, do you want those two checks to happen at the same time (in `NewConfigFromFiles` func)? @cppforlife

Reply:
what i was thinking for NewConfigFromFiles is to never return duplicate secrets. either dup secrets are exactly the same and so we can safely dedup (and only one copy would be returned), or dup secrets are different in content and so error can be returned from NewConfigFromFiles since configuration is not valid. (named look up can stay as simple as it is today just assuming that all names are unique).

Reply:
Changed the validation to be on `NewConfigFromFiles`